File: ontrack-extension-github/src/main/java/net/nemerosa/ontrack/extension/github/service/GitHubIssueServiceConfiguration.java
package net.nemerosa.ontrack.extension.github.service;

import net.nemerosa.ontrack.extension.github.GitHubIssueServiceExtension;
import net.nemerosa.ontrack.extension.github.model.GitHubEngineConfiguration;
import net.nemerosa.ontrack.extension.issues.model.IssueServiceConfiguration;

public class GitHubIssueServiceConfiguration implements IssueServiceConfiguration {

    private final GitHubEngineConfiguration configuration;

    public GitHubIssueServiceConfiguration(GitHubEngineConfiguration configuration) {
        this.configuration = configuration;
    }

    @Override
    public String getServiceId() {
        return GitHubIssueServiceExtension.GITHUB_SERVICE_ID;
    }

    @Override
    public String getName() {
        return configuration.getName();
    }
}

File: ontrack-extension-github/src/main/java/net/nemerosa/ontrack/extension/github/service/GitHubIssueServiceConfiguration.kt
package net.nemerosa.ontrack.extension.github.service

import net.nemerosa.ontrack.extension.github.GitHubIssueServiceExtension
import net.nemerosa.ontrack.extension.github.model.GitHubEngineConfiguration
import net.nemerosa.ontrack.extension.issues.model.IssueServiceConfiguration

data class GitHubIssueServiceConfiguration(
    val configuration: GitHubEngineConfiguration,
    val repository: String
) : IssueServiceConfiguration {

    override val serviceId: String = GitHubIssueServiceExtension.GITHUB_SERVICE_ID
    override val name: String = "${configuration.name}:$repository"

}